src/sentry/middleware/integrations/parsers/gitlab.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import logging
+from typing import Tuple
+
+from django.http import HttpResponse
+from django.urls import resolve
+
+from sentry.integrations.utils.scope import clear_tags_and_context
+from sentry.middleware.integrations.parsers.base import BaseRequestParser
+from sentry.models.integrations.integration import Integration
+from sentry.models.outbox import WebhookProviderIdentifier
+from sentry.services.hybrid_cloud.util import control_silo_function
+from sentry.types.integrations import EXTERNAL_PROVIDERS, ExternalProviders
+
+logger = logging.getLogger(__name__)
+
+
+class GitlabRequestParser(BaseRequestParser):
+    provider = EXTERNAL_PROVIDERS[ExternalProviders.GITLAB]
+    webhook_identifier = WebhookProviderIdentifier.GITLAB
+    _integration = None
+
+    def _get_external_id(self) -> Tuple[str, str] | HttpResponse:
```

Comment:
It looks like this logic is shared in the `webhooks.py` file for GitLab. Just to make sure that if we make a change it gets propagated to both places, could we create some functions that we just invoke in both places? It'll help DRY this code as well.

Reply:
You may be able to get around returning HttpResponses as well, so the return type is consistent, but I don't think that's a blocker, this works just fine :+1: